packages/jest-config/src/normalize.js

```diff
@@ -64,17 +64,25 @@ const setupPreset = (
   let preset;
   const presetPath = replaceRootDirInPath(options.rootDir, optionsPreset);
   const presetModule = Resolver.findNodeModule(
-    presetPath.endsWith(JSON_EXTENSION)
+    presetPath.charAt(0) === '.'
       ? presetPath
       : path.join(presetPath, PRESET_NAME),
     {
       basedir: options.rootDir,
+      extensions: PRESET_EXTENSIONS,
     },
   );
 
   try {
     // $FlowFixMe
     preset = (require(presetModule): InitialOptions);
+
+    // Force re-evaluation to support multiple projects
+    try {
+      if (presetModule) {
+        delete require.cache[require.resolve(presetModule)];
```

Comment:
Can you change this to delete before and require after? What you have here is smart, but if user code requires the preset in between, the next re-require of the preset will not actually re-evaluate the module code.